kolibri/core/content/test/test_content_app.py

```diff
@@ -506,11 +506,11 @@ def test_contentnode_tree_filtered_queryset_node(self):
         )
         self.assertEqual(response.status_code, 404)
 
-    def test_contentnode_tree_bad_pk(self):
+    def test_contentnode_tree_none_pk(self):
```

Comment:
here  for `test_contentnode_tree_bad_pk`(when pk is invalid but not None) getting 404 even with this logic 
```py
if pk is None:
            raise Http404
        else:
            try:
                UUID(pk)
            except ValueError:
                return Response(
                    {"error": "Invalid UUID format."},
                    status=status.HTTP_400_BAD_REQUEST,
                )
``` 
and i also tried it by putting this logic inside `BaseContentNodeTreeViewset` but didn't work for `pk=None`

Reply:
I don't see where you have implemented this. The else would not be there. As we want to check for both separately.
Something like this:
```
if pk is None:
    raise Http404

try:
    UUID(pk)
except ValueError:
    return Response(
        {"error": "Invalid UUID format."},
        status=status.HTTP_400_BAD_REQUEST,
    )
```

Reply:
Apologies for this, this my last commit check the new one